### PR TITLE
better setops docs

### PR DESCRIPTION
**Pull request by ringabout**
The docs
image

### CODE REVIEW

File: lib/system/setops.nim [hunk 1]
```diff
@@ -1,94 +1,91 @@
...
-  ## Returns the cardinality of the set ``x``, i.e. the number of elements
+func card*[T](x: set[T]): int {.magic: "Card".} =
+  ## Returns the cardinality of the set `x`, i.e. the number of elements
   ## in the set.
-  ##
-  ## .. code-block:: Nim
-  ##   var a = {1, 3, 5, 7}
-  ##   echo card(a) # => 4
+  runnableExamples:
+    var a = {1, 3, 5, 7}
```
Comment (timotheecour) [inline lib/system/setops.nim:43]: can you make this example more interesting eg:
```nim
var a = {1, 3, 5, 7, 5}
assert card(a) == 4 # repeated 5 doesn't count
```

File: lib/system/setops.nim [hunk 2]
```diff
@@ -1,94 +1,91 @@
...
-  ##   let
-  ##     a = {1, 2, 3}
-  ##     b = {2, 3, 4}
-  ##   echo a * b # => {2, 3}
-proc `+`*[T](x, y: set[T]): set[T] {.magic: "PlusSet", noSideEffect.}
+  runnableExamples:
+    let
+      a = {1, 2, 3}
+      b = {2, 3, 4}
+    assert a * b == {2, 3}
```
Comment (timotheecour) [inline lib/system/setops.nim]: shorter = better (+ elsewhere, eg in + and -):
```nim
assert {1, 2, 3} * {2, 3, 4} == {2, 3}
```

File: lib/system/setops.nim (hunk 1 above)
Comment (ringabout) [inline lib/system/setops.nim:43]:
```suggestion
    var a = {1, 3, 5, 7, 5}
```

File: lib/system/setops.nim [hunk 3]
```diff
@@ -1,94 +1,91 @@
...
-  ##   assert 'b' in s
-  ##
-  ## If ``in`` had been declared as ``[T](elem: T, s: set[T])`` then ``T`` would
-  ## have been bound to ``char``. But ``s`` is not compatible to type
-  ## ``set[char]``! The solution is to bind ``T`` to ``range['a'..'z']``. This
-  ## is achieved by reversing the parameters for ``contains``; ``in`` then
+  runnableExamples:
+    var s: set[range['a'..'z']] = {'a'..'c'}
+    assert s.contains('c')
+    assert 'b' in s
```
Comment (timotheecour) [inline lib/system/setops.nim:79]: add: `assert 'A' notin s`

File: lib/system/setops.nim [hunk 4]
```diff
@@ -1,94 +1,91 @@
...
-  ##   var s: set[range['a'..'z']] = {'a'..'c'}
-  ##   assert s.contains('c')
-  ##   assert 'b' in s
-  ##
-  ## If ``in`` had been declared as ``[T](elem: T, s: set[T])`` then ``T`` would
-  ## have been bound to ``char``. But ``s`` is not compatible to type
-  ## ``set[char]``! The solution is to bind ``T`` to ``range['a'..'z']``. This
-  ## is achieved by reversing the parameters for ``contains``; ``in`` then
+  runnableExamples:
+    var s: set[range['a'..'z']] = {'a'..'c'}
```
Comment (timotheecour) [inline lib/system/setops.nim:77]: how about write this instead:
```nim
  var s: set['a'..'z'] = {'a'..'c'}
  doAssert set['a'..'z'] is set[range['a'..'z']] # either works
```

File: lib/system/setops.nim (hunk 3 above)
Comment (ringabout) [reply lib/system/setops.nim:79]: This won't work
```
conversion from char to range 97..122(char) is invalid
```

**Review by timotheecour — APPROVED** [review]

File: lib/system/setops.nim (hunk 3 above)
Comment (timotheecour) [reply lib/system/setops.nim:79]: ok, this works: `assert 'd' notin s`

File: lib/system/setops.nim [hunk 5]
```diff
@@ -1,94 +1,83 @@
...
-  ## Returns the cardinality of the set ``x``, i.e. the number of elements
+func card*[T](x: set[T]): int {.magic: "Card".} =
+  ## Returns the cardinality of the set `x`, i.e. the number of elements
   ## in the set.
-  ##
-  ## .. code-block:: Nim
-  ##   var a = {1, 3, 5, 7}
-  ##   echo card(a) # => 4
+  runnableExamples:
+    var a = {1, 3, 5, 7, 5}
```
Comment (Clyybber) [inline lib/system/setops.nim]: Is this duplicate 5 there by accident?
```suggestion
    var a = {1, 3, 5, 7}
```

Comment (ringabout) [reply lib/system/setops.nim]: It is intended:
https://github.com/nim-lang/Nim/pull/16236#discussion_r534601784

File: lib/system/setops.nim [hunk 6]
```diff
@@ -1,94 +1,83 @@
...
+    assert card(a) == 4
```
Comment (ringabout) [inline lib/system/setops.nim:44]:
```suggestion
    assert card(a) == 4 # repeated 5 doesn't count
```

File: lib/system/setops.nim [hunk 7]
```diff
@@ -1,94 +1,83 @@
...
-  ##   assert 'b' in s
-  ##
-  ## If ``in`` had been declared as ``[T](elem: T, s: set[T])`` then ``T`` would
-  ## have been bound to ``char``. But ``s`` is not compatible to type
-  ## ``set[char]``! The solution is to bind ``T`` to ``range['a'..'z']``. This
-  ## is achieved by reversing the parameters for ``contains``; ``in`` then
+  runnableExamples:
+    var s: set[range['a'..'z']] = {'a'..'c'}
+    assert s.contains('c')
+    assert 'b' in s
```
Comment (ringabout) [inline lib/system/setops.nim:79]:
```suggestion
    assert 'b' in s
    assert 'd' notin s
```

File: lib/system/setops.nim (hunk 6 above)
Comment (Clyybber) [reply lib/system/setops.nim:44]: That is a general property of sets though, not of `card`.

File: lib/system/setops.nim [hunk 8]
```diff
@@ -1,94 +1,84 @@
...
+func card*[T](x: set[T]): int {.magic: "Card".} =
+  ## Returns the cardinality of the set `x`, i.e. the number of elements
   ## in the set.
-  ##
-  ## .. code-block:: Nim
-  ##   var a = {1, 3, 5, 7}
-  ##   echo card(a) # => 4
+  runnableExamples:
+    var a = {1, 3, 5, 7, 5}
+    assert card(a) == 4 # repeated 5 doesn't count
```
Comment (Clyybber) [inline lib/system/setops.nim]:
```suggestion
    var a = {1, 3, 5, 7}
    assert card(a) == 4
```
see https://github.com/nim-lang/Nim/pull/16236#discussion_r535191726

File: lib/system/setops.nim (hunk 6 above)
Comment (ringabout) [reply lib/system/setops.nim:44]: I will make two examples and one is used to promote the properties of `set`.

File: lib/system/setops.nim (hunk 8 above)
Comment (ringabout) [reply lib/system/setops.nim]: I will make two examples and one is used to promote the properties of set.

**Pull request merged**